### PR TITLE
8273410: IR verification framework fails with "Should find method name in validIrRulesMap"

### PR DESCRIPTION
**Pull request by chhagedorn**
The IR framework treated a `@Check` method as `@Test` method instead of the `@Test` method itself at IR matching time resulting in an internal framework exception.

While writing some tests for checked test I've noticed that a missing `@Arguments` annotation is not reported as `TestFormatException` but with a `RuntimeException` later when invoking the method in question. I also added the missing check for it.

Thanks,
Christian

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273410](https://bugs.openjdk.java.net/browse/JDK-8273410): IR verification framework fails with "Should find method name in validIrRulesMap"

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to e2a5d83edc38c6b46768900c67078df1c9940696
 * [Nils Eliasson](https://openjdk.java.net/census#neliasso) (@neliasso - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5678/head:pull/5678` \
`$ git checkout pull/5678`

Update a local copy of the PR: \
`$ git checkout pull/5678` \
`$ git pull https://git.openjdk.java.net/jdk pull/5678/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5678`

View PR using the GUI difftool: \
`$ git pr show -t 5678`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5678.diff">https://git.openjdk.java.net/jdk/pull/5678.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestCheckedTests.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    @Check(test = "testBad2")
+    public void sameName(int retValue) {
+        if (retValue != 3) {
+            throw new RuntimeException("must be 3");
+        }
+    }
+
+
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestCheckedTests.java:201]: Double newline.

File: test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestCheckedTests.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+package ir_framework.tests;
+
+import compiler.lib.ir_framework.*;
+import compiler.lib.ir_framework.driver.IRViolationException;
+import compiler.lib.ir_framework.driver.TestVMException;
+import jdk.test.lib.Asserts;
+
+/*
+ * @test
+ * @requires vm.debug == true & vm.compMode != "Xint" & vm.compiler2.enabled & vm.flagless
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/testlibrary_tests/ir_framework/tests/TestCheckedTests.java:33]: You might want to add an `@bug` tag.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by chhagedorn** [comment]
Thanks Tobias for your review!

**Comment by chhagedorn** [comment]
Anyone for a second review?

**Comment by chhagedorn** [comment]
Thanks Nils for your review!

**Pull request closed without merging**